### DISCUSSION TITLE
Zoneless components for Angular 18

### DISCUSSION THREAD

**Discussion by hakimio** [taiga-family/taiga-ui#7221] [Ideas]
Angular 18 will introduce zoneless signal components and "Angular Material" component library has already done the required migrations to support zoneless. Do you have any plans to support "zoneless Angular"?

Some of the PRs merged to migrate Angular Material:
https://github.com/angular/components/pull/28332
https://github.com/angular/components/pull/28782
https://github.com/angular/components/pull/28783

PS: right now "Taiga UI" doesn't work correctly with the private `ɵprovideZonelessChangeDetection()` API but that might be quite different from the real zoneless experience which will be delivered in Angular 18.

**Comment by waterplea** [comment]
Thank you for bringing that up. We will see if we can do the required changes in Angular 16 which will be our base version for Taiga UI 4.0

**Comment by hakimio** [comment]
zoneless scheduler PRs were just merged for Angular 18:
https://github.com/angular/angular/pull/55102
https://github.com/angular/angular/pull/55252

And released in v18.0.0-next.5:
https://github.com/angular/angular/releases/tag/18.0.0-next.5

**Reply by hakimio** [reply]
First article about the new `provideExperimentalZonelessChangeDetection()` provider:
https://justangular.com/blog/a-new-era-for-angular-zoneless-change-detection

Native async/await, when using zoneless CD, is coming too:
https://github.com/angular/angular-cli/pull/27486

Will fix some of the `zone.js` related promise issues:
https://github.com/angular/angular/issues/47562
https://github.com/angular/angular/issues/51328

**Comment by hakimio** [comment]
Angular folks have published [Requirements for Zoneless compatibility](https://github.com/angular/angular/blob/main/adev/src/content/guide/zoneless.md#requirements-for-zoneless-compatibility).
Basically **do** use `ChangeDetectionStrategy.OnPush` and **don't** use `NgZone.onMicrotaskEmpty`, `NgZone.onUnstable`, `NgZone.isStable`, or `NgZone.onStable`.

**Comment by hakimio** [comment]
Just tested Taiga `v3` with Angular 18 `provideExperimentalZonelessChangeDetection()`.
All functionality works as expected but there are a lot of issues with Taiga component styling. For example:
- Taiga `Tabs` component active tab doesn't have underline
- Taiga `button` and `link` with `tuiMode="onLight"` directive has incorrect style _(mode directive removed in `v4` in favor of `tuiAppearance="icon"`)_
- `Accordion` collapse animation is broken
- `Dialog` in mobile mode is not shown in fullscreen

Hopefully Taiga `v4` will do better.

**Comment by hakimio** [comment]
Tested with Taiga `4.6.0` and Angular `18.2.3`:
- `Tabs` active tab underline is fixed now
- `tuiMode="onLight"` no longer relevant since it was removed. `appearance="icon"` works fine.
- ~~`Accordion` collapse animation is still broken (might be related to https://github.com/taiga-family/taiga-ui/issues/8973)~~
- ~~`Dialog` in mobile mode is still not shown in fullscreen~~

**UPDATE:** the old accordion component was deprecated and [the new one](https://taiga-ui.dev/components/accordion) works in `zoneless` mode.
**UPDATE 02/06/2025:** the `Dialog` seems to be working correctly now too with the latest Taiga version. Can't find any other issues now.
